RecoLocalTracker/SiPixelClusterizer/plugins/SiPixelClusterProducer.cc

```diff
@@ -120,8 +133,11 @@
 
     // Step D: write output to file
     output->shrink_to_fit();
-    e.put(std::move(output));
+    e.put(tPutPixelClusters, std::move(output));
 
+    if(makeDummyDigis_) {
+      e.emplace(tPutPixelDigi);
+    }
```

Comment:
This is just stupid (but was easiest to proceed). I'll try to think some better solution.

(ultimately this tension comes back to legacy CPU workflow producing (CPU) digis and clusters by two separate modules, and GPU workflow producing the CPU digis and clusters by a single module)

Reply:
I'll try to see if I could easily make the SwitchProducer to take an EDAlias as one of the cases.

If that doesn't work out, the easiest short-term solution would be to split the filling of digis and clusters
https://github.com/cms-patatrack/cmssw/blob/0800aac1eeea41ccb73826a4315c82a0f1a6f8cd/RecoLocalTracker/SiPixelClusterizer/plugins/SiPixelClustersFromSoA.cc#L87-L155
to two separate modules.

The ultimate solution would likely be changing the CPU side as well to make digis and clusters in a single module (like in GPU), but I presume that would take some time.

Reply:
> ultimately this tension comes back to legacy CPU workflow producing (CPU) digis and clusters by two separate modules, and GPU workflow producing the CPU digis and clusters by a single module

why is it a problem? at least, form the discussion we had in the past I understood it was fine to have the CPU and GPU versions of a module produce collections with different "granularity".

Reply:
#25907 extends SwitchProducer to switch between EDProducer(s) and EDAlias(es). Here is how that functionality could be used on top of this PR https://github.com/makortel/cmssw/commit/470ee3697afa2feeea7eeaa7dc0a5749c073f901 (I'm reluctant to include it in this PR for now because `git cms-checkdeps -a` on #25907 brings the universe).

Reply:
> > ultimately this tension comes back to legacy CPU workflow producing (CPU) digis and clusters by two separate modules, and GPU workflow producing the CPU digis and clusters by a single module
> 
> why is it a problem? at least, form the discussion we had in the past I understood it was fine to have the CPU and GPU versions of a module produce collections with different "granularity".

That's the point, it should not be a problem. In this PR (on top of 10_5_0_pre1) it still is, because
* CPU (legacy) clusterizer produces `SiPixelClusterCollectionNew`
* GPU "CPU SoA to legacy" converter produces `SiPixelClusterCollectionNew` and `edm::DetSetVector<PixelDigi>`
* SwitchProducer requires all cases to produce the same set of products, so the above setup does not work (gives an exception at run time)

That leads to allowing EDAlias as a case in the SwitchProducer (https://github.com/cms-sw/cmssw/pull/25907), since that allows to keep the current structure on CPU and GPU side, and the enforcement that a given module label always has the same products produced (within a SwitchProducer).